typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -769,6 +775,17 @@
           [_ type])]))
     #f))
 
+;; True if the arities `arrs` are what we'd expect from a struct predicate
+(define (any->bool? arrs)
+  (match arrs
+   [(list (arr: (list Univ:) x #f #f '()))
```

Comment:
Then figure out where to `require` it from. Repeating the definition of booleans isn't the right solution.